src/main/java/com/rackspace/ceres/app/services/DataWriteService.java

```diff
@@ -138,4 +143,27 @@ private void cleanTags(Map<String, String> tags) {
         .retryWhen(appProperties.getRetryInsertDownsampled().build())
         .checkpoint();
   }
+
+    private Mono<?> storeMetricGroup(String tenant, Metric metric) {
+        String metricGroup = metric.getTags().get(LABEL_METRIC_GROUP);
+        String metricName = metric.getMetric();
+        Flux<Row> rows = metadataService.getRowsMetricNamesFromMetricGroup(tenant, metricGroup);
```

Comment:
Can we directly use Update query in cassandra here instead of first getting and inserting?
For reference:- https://docs.datastax.com/en/cql-oss/3.3/cql/cql_using/useInsertList.html

Reply:
Yes, we can, the problem is that there doesn't seem to be any support for `NOT CONTAINS` in cassandra. I will see if I can find a solution around that.